packages/langium/src/workspace/ast-descriptions.ts

```diff
@@ -25,30 +26,40 @@ export interface AstNodeDescriptionProvider {
      * the contents of a document and during scope computation.
      *
      * @param node An AST node.
-     * @param name The name to be used to refer to the AST node. Typically this is determined by the
+     * @param name The name to be used to refer to the AST node. By default, this is determined by the
      *     `NameProvider` service, but alternative names may be provided according to the semantics
      *     of your language.
      * @param document The document containing the AST node. If omitted, it is taken from the root AST node.
      */
-    createDescription(node: AstNode, name: string, document?: LangiumDocument): AstNodeDescription;
+    createDescription(node: AstNode, name: string | undefined, document?: LangiumDocument): AstNodeDescription;
 
 }
 
 export class DefaultAstNodeDescriptionProvider implements AstNodeDescriptionProvider {
 
     protected readonly astNodeLocator: AstNodeLocator;
+    protected readonly nameProvider: NameProvider;
 
     constructor(services: LangiumServices) {
         this.astNodeLocator = services.workspace.AstNodeLocator;
+        this.nameProvider = services.references.NameProvider;
     }
 
-    createDescription(node: AstNode, name: string, document: LangiumDocument = getDocument(node)): AstNodeDescription {
+    createDescription(node: AstNode, name: string | undefined, document: LangiumDocument = getDocument(node)): AstNodeDescription {
+        name ??= this.nameProvider.getName(node);
+        const path = this.astNodeLocator.getAstNodePath(node);
+        if (name === undefined) {
```

Comment:
I think it's ok to check for `!name` here because an empty string is not a name.